packages/modular-scripts/src/esbuild-scripts/start/index.ts

```diff
@@ -143,13 +146,55 @@ class DevServer {
     await this.startEsbuildServer();
     await this.hostRuntime();
 
+    const { SSL_CRT_FILE, SSL_KEY_FILE, HTTPS } = process.env;
+    const isHttps = HTTPS === 'true';
+
+    let selfSignedCert: Buffer | undefined;
+    if (isHttps) {
+      selfSignedCert = await generateSelfSignedCert();
```

Comment:
I've had to restructure things a bit in 8c8e9c1190730acf8356d7a8a9db47cbd75047b7 after realising that the websocket server also needs to be booted with the correct server instance (whereas before, it would boot the websocket server without https, right away).

I have resolved this particular inefficiency as part of that change.